activemodel/lib/active_model/normalization.rb

```diff
@@ -0,0 +1,156 @@
+# frozen_string_literal: true
+
+module ActiveModel # :nodoc:
+  module Normalization
+    extend ActiveSupport::Concern
+
+    included do
+      include ActiveModel::Dirty
+      include ActiveModel::Validations::Callbacks
+
+      class_attribute :normalized_attributes, default: Set.new
+
+      before_validation :normalize_changed_in_place_attributes
+    end
+
+    # Normalizes a specified attribute using its declared normalizations.
+    #
+    # ==== Examples
+    #
+    #   class User
+    #     include ActiveModel::Model
+    #     include ActiveModel::Attributes
+    #     include ActiveModel::Normalization
+    #
+    #     attribute :email, :string
+    #
+    #     normalizes :email, with: -> email { email.strip.downcase }
+    #   end
+    #
+    #   legacy_user = User.find(1)
```

Comment:
`User.find` wouldn't be defined here.  Maybe just use a placeholder, like `# ...`?  Or an obviously made up function, like `load_user(1)`?